pkg/virt-handler/heartbeat/ksm.go

```diff
@@ -249,33 +249,27 @@ func getIntParam(node *v1.Node, param string, defaultValue, lowerBound, upperBou
 // will set the outcome value to the n.KSM struct
 // If the node labels match the selector terms, the ksm will be enabled.
 // Empty Selector will enable ksm for every node
-func handleKSM(node *v1.Node, clusterConfig *virtconfig.ClusterConfig) (bool, bool) {
-	available, running := loadKSM()
+func handleKSM(node *v1.Node, clusterConfig *virtconfig.ClusterConfig) (ksmLabelValue, ksmEnabledByUs bool) {
+	available, enabled := loadKSM()
 	if !available {
-		return running, false
+		return false, false
 	}
 
 	ksmConfig := clusterConfig.GetKSMConfiguration()
-	if ksmConfig == nil {
-		if disableKSM(node, running) {
-			return false, false
-		} else {
-			return running, false
-		}
+	if ksmConfig == nil && enabled {
+		disableKSM(node)
+		return false, false
 	}
 
 	selector, err := metav1.LabelSelectorAsSelector(ksmConfig.NodeLabelSelector)
 	if err != nil {
 		log.DefaultLogger().Errorf("An error occurred while converting the ksm selector: %s", err)
-		return running, false
+		return false, false
 	}
 
-	if !selector.Matches(labels.Set(node.ObjectMeta.Labels)) {
-		if disableKSM(node, running) {
-			return false, false
-		} else {
-			return running, false
-		}
+	if !selector.Matches(labels.Set(node.ObjectMeta.Labels)) && enabled {
+		disableKSM(node)
+		return false, false
```

Comment:
why do we need the `&& enabled` ?
I think that we should return from this func even if it is not enabled

Reply:
You are right, but we also don't want to write always the same value, when ksm is disabled.
Will adjust! Thank you